napari/_qt/widgets/magic_gui_base_widget.py

```diff
@@ -0,0 +1,60 @@
+from qtpy.QtCore import QObject, Signal
+
+
+class BaseMagicSetting(QObject):
+    """Base class that helps use Magic GdUI widgets in the json schema widget
+    buider.
+    """
+
+    valueChanged = Signal(dict)
+
+    def get_mgui(self):
+        """Get magic gui widget."""
+        raise NotImplementedError()
+
+    def __init__(self, description=None):
+        super().__init__()
+        self._widget = self.get_mgui()
+        self._description = description
+
+        @self._widget.changed.connect
+        def _call_magic_gui(value: dict = {}):
+            self.valueChanged.emit(value())
```

Comment:
Why is this not QWidget? 

```suggestion
class BaseMagicSetting(QWidget):
    """Base class that helps use Magic GdUI widgets in the json schema widget
    buider.
    """

    valueChanged = Signal(dict)

    def get_mgui(self):
        """Get magic gui widget."""
        raise NotImplementedError()

    def __init__(self, description=None):
        super().__init__()
        self._widget = self.get_mgui()
        self._description = description
        
        self.setLayout(QHBoxLayout())
        self.layout().addWidget(self._widget.native)

        self._widget.changed.connect(self._call_magic_gui)

    def _call_magic_gui(self, value: dict = None):
        if self.value is None:
            value = {}
        self.valueChanged.emit(value)
```

Reply:
I made these changes.  Thank you.